graphql/src/execution/execution.rs

```diff
@@ -244,10 +263,36 @@ pub fn execute_root_selection_set(
         )?);
     }
 
-    if let Some(key) = key {
-        // Insert into the cache.
-        let mut cache = QUERY_CACHE.lock().unwrap();
-        cache.insert(key, values.clone());
+    // Check if this query should be cached.
```

Comment:
The code for writing to the cache is a bit tricky, and probably could be written in different, better ways. But as long as we can agree that the current form is correct, performs well even if there are micro-optimizations that could be made and reads ok, I'm happy with its current form.